app/assets/javascripts/trips.controller.js

```diff
@@ -10,6 +10,8 @@
   ]);
 
   function tripsIndexCtrlFunction(TripFactory, ItineraryFactory, DestinationFactory) {
+// JGZ - I think in terms of "optimizing" this, I don't think this is hitting the DB too many times
+// many-many is hard, so it makes sense that you would need to query the DB more often.
     this.trips = TripFactory.all
```

Comment:
side note. was talking to Nick about this the other day, and he was working through a similar issue and figured out a way to minimize the number of queries on the backend. will try to find some example at some point and slack to you guys